File: temp3/app/src/main/java/com/example/aradhya/MainActivity.java
package com.example.aradhya;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
    public void ChangeText(View view){
        display(0);
    }
    public void display(int s)
    {
        TextView text = (TextView) findViewById(R.id.anmolview);
        text.setText("The ans is "+s);
    }
}

File: Learning Android App/retrofit_skylark/app/src/main/java/com/example/retrofit_skylark/MainActivity.kt
package com.example.retrofit_skylark

import android.annotation.SuppressLint
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.TextView
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import dagger.hilt.android.AndroidEntryPoint


@AndroidEntryPoint
class MainActivity : AppCompatActivity() {


    private lateinit var viewModel: ViewModelClass by viewModels()

    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val text = findViewById<TextView>(R.id.textView)
        val button = findViewById<TextView>(R.id.button)

        viewModel = ViewModelProvider(this)[ViewModelClass::class.java]

        viewModel.getLiveData().observe(this, Observer {
            Log.d("resp values", it)
            if (it != "") {
                text.text = it
            } else {
                text.text = "empty"
            }
        })

        button.setOnClickListener {
//            here we can show the loading state as we are making the http request in background
            viewModel.getData()
        }

    }
}